src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,424 @@
+   * Firefox Mobile, Opera Mobile. And can also work well on Safari Mobile with a "-webkit-" prefix.
+   * It won't always work on IE phone.
+   *
+   * The implementation references the compatibility checking in Modernizer
+   * (https://github.com/Modernizr/Modernizr/blob/master/feature-detects/css/positionsticky.js).
+   */
+  getSupportList(): Array<string> {
+    let prefixTestList = ['', '-webkit-', '-ms-', '-moz-', '-o-'];
+    let supportList: Array<string> = new Array<string>();
+    let stickyText = '';
+    for (let i = 0; i < prefixTestList.length; i++ ) {
+      stickyText = 'position:' + prefixTestList[i] + 'sticky;';
+      // Create a DOM to check if the browser support current prefix for sticky-position.
+      let div = document.createElement('div');
+      let body = document.body;
+      div.style.cssText = 'display:none;' + stickyText;
+      body.appendChild(div);
+      let isSupport = /sticky/i.test(window.getComputedStyle(div).position);
+      body.removeChild(div);
+      div = null;
+      if (isSupport == true) {
+        supportList.push(prefixTestList[i]);
+      }
+    }
+    return supportList;
+  }
+
+  /**
+   * Get the first element from this._supportList. Set it as a prefix of
+   * sticky positioning.
+   *
+   * If the this._supportList is empty, which means the browser does not support
+   * sticky positioning. Set isStickyPositionSupported as 'true' and use the original
+   * implementation of sticky-header.
+   */
+  setStrategyAccordingToCompatibility(): void {
+    let supportList = this.getSupportList();
+    if (supportList.length === 0) {
+      this.isStickyPositionSupported = false;
+    } else {
+      // Only need supportList[0], Because supportList contains all the prefix
+      // that can make sticky positioning work in the current browser.
+      // We only need to get one prefix and make position: prefix + 'sticky',
+      // then sticky position will work.
+      let prefix: string = supportList[0];
+
+      this.element.style.top = '0px';
+      this.element.style.position = prefix + 'sticky';
+    }
+  }
+
+  attach() {
+    this._onScrollSubscription = Observable.fromEvent(this.upperScrollableContainer, 'scroll')
+      .debounceTime(DEBOUNCE_TIME).subscribe(() => this.defineRestrictionsAndStick());
+
+    // Have to add a 'onTouchMove' listener to make sticky header work on mobile phones
+    this._onTouchSubscription = Observable.fromEvent(this.upperScrollableContainer, 'touchmove')
+      .debounceTime(DEBOUNCE_TIME).subscribe(() => this.defineRestrictionsAndStick());
+
+    this._onResizeSubscription = Observable.fromEvent(this.upperScrollableContainer, 'resize')
+      .debounceTime(DEBOUNCE_TIME).subscribe(() => this.onResize());
+  }
+
+  onScroll(): void {
+    this.defineRestrictionsAndStick();
+  }
+
+  onTouchMove(): void {
+    this.defineRestrictionsAndStick();
+  }
+
+  onResize(): void {
+    this.defineRestrictionsAndStick();
```

Comment:
Because I need to judge whether the current header need to be stuck. Calling `defineRestrictionsAndStick()` to see the current state of the header. If the current header is stuck, we need to unstuck it and then stuck it again to reshape the sticky header. And if the current header is not stuck, we will do nothing. Just let it being scrolled up or down.